File: lib/backend/supabase/database/tables/financial_cards.dart
import '../database.dart';

class FinancialCardsTable extends SupabaseTable<FinancialCardsRow> {
  @override
  String get tableName => 'financial_cards';

  @override
  FinancialCardsRow createRow(Map<String, dynamic> data) =>
      FinancialCardsRow(data);
}

class FinancialCardsRow extends SupabaseDataRow {
  FinancialCardsRow(super.data);

  @override
  SupabaseTable get table => FinancialCardsTable();

  int get id => getField<int>('id')!;
  set id(int value) => setField<int>('id', value);

  DateTime get createdAt => getField<DateTime>('created_at')!;
  set createdAt(DateTime value) => setField<DateTime>('created_at', value);

  String get createdBy => getField<String>('created_by')!;
  set createdBy(String value) => setField<String>('created_by', value);

  DateTime? get updatedAt => getField<DateTime>('updated_at');
  set updatedAt(DateTime? value) => setField<DateTime>('updated_at', value);

  String? get updatedBy => getField<String>('updated_by');
  set updatedBy(String? value) => setField<String>('updated_by', value);

  String get cardHolderName => getField<String>('card_holder_name')!;
  set cardHolderName(String value) =>
      setField<String>('card_holder_name', value);

  String get cardNumber => getField<String>('card_number')!;
  set cardNumber(String value) => setField<String>('card_number', value);

  String? get cardProviderName => getField<String>('card_provider_name');
  set cardProviderName(String? value) =>
      setField<String>('card_provider_name', value);

  String? get cardType => getField<String>('card_type');
  set cardType(String? value) => setField<String>('card_type', value);

  String? get cvv => getField<String>('cvv');
  set cvv(String? value) => setField<String>('cvv', value);

  String? get expiryDate => getField<String>('expiry_date');
  set expiryDate(String? value) => setField<String>('expiry_date', value);

  String? get issueDate => getField<String>('issue_date');
  set issueDate(String? value) => setField<String>('issue_date', value);

  String get name => getField<String>('name')!;
  set name(String value) => setField<String>('name', value);

  String? get note => getField<String>('note');
  set note(String? value) => setField<String>('note', value);

  String? get pin => getField<String>('pin');
  set pin(String? value) => setField<String>('pin', value);
}

File: rust/src/models/financial_cards.rs
use chrono::NaiveDateTime;
use crudlf_derive::{SqliteDelete, SqliteInsert, SqliteListFilter, SqliteSelect, SqliteUpdate};
use flutter_rust_bridge::frb;
use serde::{Deserialize, Serialize};
use sqlx::FromRow;

#[derive(
    Debug,
    Clone,
    PartialEq,
    Serialize,
    Deserialize,
    FromRow,
    SqliteInsert,
    SqliteSelect,
    SqliteUpdate,
    SqliteDelete,
    SqliteListFilter,
)]
#[crudlf(table_name = "financial_cards")]
#[frb(dart_metadata=("freezed"))]
pub struct FinancialCard {
    #[serde(rename = "id")]
    #[serde(skip_serializing_if = "Option::is_none")]
    pub id: Option<String>,

    #[serde(rename = "created_at")]
    #[serde(skip_serializing_if = "Option::is_none")]
    pub created_at: Option<NaiveDateTime>,

    #[serde(rename = "created_by")]
    #[serde(skip_serializing_if = "Option::is_none")]
    pub created_by: Option<String>,

    #[serde(rename = "updated_at")]
    #[serde(skip_serializing_if = "Option::is_none")]
    pub updated_at: Option<NaiveDateTime>,

    #[serde(rename = "updated_by")]
    #[serde(skip_serializing_if = "Option::is_none")]
    pub updated_by: Option<String>,

    #[serde(rename = "card_holder_name")]
    pub card_holder_name: String,

    #[serde(rename = "card_number")]
    pub card_number: String,

    #[serde(rename = "card_provider_name")]
    #[serde(skip_serializing_if = "Option::is_none")]
    pub card_provider_name: Option<String>,

    #[serde(rename = "card_type")]
    #[serde(skip_serializing_if = "Option::is_none")]
    pub card_type: Option<String>,

    #[serde(rename = "cvv")]
    #[serde(skip_serializing_if = "Option::is_none")]
    pub cvv: Option<String>,

    #[serde(rename = "expiry_date")]
    #[serde(skip_serializing_if = "Option::is_none")]
    pub expiry_date: Option<String>,

    #[serde(rename = "issue_date")]
    #[serde(skip_serializing_if = "Option::is_none")]
    pub issue_date: Option<String>,

    #[serde(rename = "name")]
    pub name: String,

    #[serde(rename = "note")]
    #[serde(skip_serializing_if = "Option::is_none")]
    pub note: Option<String>,

    #[serde(rename = "pin")]
    #[serde(skip_serializing_if = "Option::is_none")]
    pub pin: Option<String>,
}
